app/services/stripe_service/api/stripe_api_wrapper.rb

```diff
@@ -0,0 +1,273 @@
+            first_name: account_info[:first_name],
+            last_name: account_info[:last_name],
+            address: {
+              city: account_info[:address_city],
+              state: account_info[:address_state],
+              country: account_info[:address_country],
+              postal_code: account_info[:address_postal_code],
+              line1: account_info[:address_line1],
+            },
+            dob: {
+              day: account_info[:birth_date].day,
+              month: account_info[:birth_date].month,
+              year: account_info[:birth_date].year,
+            },
+            ssn_last_4: (account_info[:address_country] == 'US' ? account_info[:ssn_last_4] : nil),
+          },
+
+          tos_acceptance: {
+            date: account_info[:tos_date].to_i,
+            ip: account_info[:tos_ip],
+          }
+        }.merge(payout_mode))
+      end
+    end
+
+    def check_balance(community)
+      with_stripe_payment_config(community) do |payment_settings|
+        Stripe::Balance.retrieve
+      end
+    rescue => e
+      nil
+    end
+
+    def create_bank_account(community, account_info)
+      with_stripe_payment_config(community) do |payment_settings|
+        stripe_account = Stripe::Account.retrieve account_info[:stripe_seller_id]
+        routing = account_info[:bank_routing_number].present? ? { routing_number: account_info[:bank_routing_number] } : {}
+        data = {
+          external_account: {
+            object: 'bank_account',
+            account_number: account_info[:bank_account_number],
+            currency:       account_info[:bank_currency],
+            country:        account_info[:bank_country],
+            account_holder_name: account_info[:bank_holder_name],
+            account_holder_type: 'individual'
+          }.merge(routing)
+        }
+
+        stripe_account.external_accounts.create(data)
+      end
+    end
+
+    def get_account_balance(community, account_id)
+      with_stripe_payment_config(community) do |payment_settings|
+        Stripe::Balance.retrieve({stripe_account: account_id})
+      end
+    end
+
+    def perform_payout(community, account_id, amount_cents, currency)
+      with_stripe_payment_config(community) do |payment_settings|
+        Stripe::Payout.create({amount: amount_cents, currency: currency}, {stripe_account: account_id})
+      end
+    end
+
+    def perform_transfer(community, account_id, amount_cents, amount_currency, initial_amount, charge_id)
+      with_stripe_payment_config(community) do |payment_settings|
+        charge = Stripe::Charge.retrieve(charge_id)
+        balance_txn = Stripe::BalanceTransaction.retrieve(charge.balance_transaction)
+        balance_currency = balance_txn.currency.upcase
+        # when platform balance is, say in EUR, but prices are in USD, recalc amount
+        fixed_amount = balance_currency == amount_currency ? amount_cents : (amount_cents * 1.0 / initial_amount * balance_txn.amount).to_i
```

Comment:
it is not relevant since we force the sellers to have bank accounts in the same currency as marketplace, and by default use destination payments